core/container.go

```diff
@@ -92,13 +92,19 @@ func NewContainer(id ContainerID, pipeline pipeline.Path, platform specs.Platfor
 // WithXXX method.
 func (container *Container) Clone() *Container {
 	cp := *container
-	cp.Mounts = clone(cp.Mounts)
-	cp.Secrets = clone(cp.Secrets)
-	cp.Sockets = clone(cp.Sockets)
-	cp.Ports = clone(cp.Ports)
+	cp.Config.ExposedPorts = cloneMap(cp.Config.ExposedPorts)
```

Comment:
Yeah, worth considering. I figured these would be slow-moving enough to not need it, but it's definitely a source for tech debt with infrequent but potent impact, which is always nice to avoid. Probably won't do it in this PR, but I'll keep an eye on it!